src/poprox_storage/scrape_articles.py

```diff
@@ -0,0 +1,264 @@
+df = pd.DataFrame(articles)
+print(df.head())
+
+# Add a time stamp to the file name
+now = datetime.datetime.now()
+now = now.strftime("%Y-%m-%d_%H-%M-%S")
+
+# # Save to the folder "articles"
+# path_to_save = f'scraped-articles/apnews_articles_{now}.csv'
+# df.to_csv(path_to_save, index=False)
+# print(f'Articles saved to {path_to_save}')
+
+# Database Connection and Table Creation
+
+# Database connection parameters
+conn_params = {
+    "dbname": "poprox",
+    "user": "postgres",
+    "password": "thisisapoproxpassword",
+    "host": "127.0.0.1",
+    "port": "5433",
+}
+
+# Connect to the PostgreSQL server
+try:
+    conn = psycopg2.connect(**conn_params)
+    print("Connection successful!")
+
+    # Create a cursor object
+    cur = conn.cursor()
+
+    # Check if the 'scraped_articles' table exists
+    cur.execute("""
+        SELECT EXISTS (
+            SELECT 1
+            FROM information_schema.tables
+            WHERE table_schema = 'public'
+            AND table_name = 'scraped_articles'
+        );
+    """)
+
+    table_exists = cur.fetchone()[0]
+
+    # If the 'scraped_articles' table doesn't exist, create it
+    if not table_exists:
+        cur.execute("""
+            CREATE TABLE scraped_articles (
+                title VARCHAR,
+                description VARCHAR,
+                url VARCHAR PRIMARY KEY,
+                section VARCHAR,
+                level VARCHAR,
+                image_url VARCHAR,
+                scraped_at TIMESTAMP
+            );
+        """)
+        print("Table 'scraped_articles' created successfully!")
+    else:
+        print("Table 'scraped_articles' already exists.")
+
+    # Query to get the table schema
+    cur.execute("""
+        SELECT column_name, data_type
+        FROM information_schema.columns
+        WHERE table_name = 'scraped_articles';
+    """)
+
+    # Fetch all the results
+    schema = cur.fetchall()
+
+    # Print the schema of the table
+    print("Schema of the 'scraped_articles' table:")
+    for column in schema:
+        print(f"Column: {column[0]}, Data Type: {column[1]}")
+
+    # Insert the data into the table
+    for idx, row in df.iterrows():
+        try:
+            cur.execute(
+                """
+                INSERT INTO scraped_articles (title, description, url, section, level, image_url, scraped_at)
+                VALUES (%s, %s, %s, %s, %s, %s, %s)
+                ON CONFLICT (url) DO NOTHING;  -- To avoid duplicate entries based on the URL
+            """,
+                (
+                    row["title"],
+                    row["description"],
+                    row["url"],
+                    row["section"],
+                    row["level"],
+                    row["image_url"],
+                    row["scraped_at"],
+                ),
+            )
+            print(f"Inserted row {idx + 1} into 'scraped_articles' table.")
+        except Exception as e:
+            print(f"Error inserting row {idx + 1}: {e}")
```

Comment:
This part looks like it could become a `store_scraped_article` method on a new `DbScrapedArticleRepository` class. Here's [an example](https://github.com/CCRI-POPROX/poprox-storage/blob/main/src/poprox_storage/repositories/clicks.py#L37-L48) of a `store` method that does something similar using `sqlalchemy`'s API (instead of raw SQL) that you can use as a template.